environment.go

```diff
@@ -142,13 +142,13 @@ func NewEnvironment(ctx context.Context, confPath string, db *DBSettings) (Envir
 	e := &envState{
 		ctx:     ctx,
 		senders: map[SenderKey]send.Sender{},
-		closers: map[string]func(context.Context) error{
-			"root-context": func(_ context.Context) error {
-				cancel()
-				return nil
-			},
-		},
 	}
+	defer func() {
```

Comment:
because the root context should be the last thing canceled (we want queues to be closed and finish closing before we cancel the root context, otherwise the work aborts in a weird order.)